core/src/main/java/com/graphhopper/routing/profiles/HazmatTunnel.java

```diff
@@ -0,0 +1,33 @@
+package com.graphhopper.routing.profiles;
+
+/**
+ * Defines the degree of restriction for the transport of hazardous goods through tunnels.<br>
+ * If not tagged it will be {@link #A}
+ * 
+ * @see https://wiki.openstreetmap.org/wiki/Key:hazmat#Tunnel_restrictions
+ */
+public enum HazmatTunnel {
+    /** driving with any dangerous goods allowed */
+    A("A"),
+    /** no goods with very large explosion range */
+    B("B"),
+    /** no goods with large explosion or poisoning range */
+    C("C"),
+    /** no goods which threaten a large explosion, poisoning or fire */
+    D("D"),
+    /** forbids all dangerous goods except: UN 2919,3291, 3331, 3359, 3373 */
+    E("E");
```

Comment:
I'm not sure if we should introduce this EncodedValue. The world wide the usage is under 500 ways:

```
290 | hazmat:E=no 
24  | hazmat:D=no 
59  | hazmat:C=no 
115 | hazmat:B=no 
15  | hazmat:A=no 
```

See https://taginfo.openstreetmap.org/keys/hazmat%3AA#values

Reply:
The numbers are low but we're talking about tunnels. They are huge shortcuts and have drastic effect on the final route. Just imagine if your vehicle isn't allowed to use the Gotthard Tunnel.

Reply:
There can be many big issues when certain road properties are not considered, this property is nothing special. And it is always possible to include it even if this is not part of GH core.

The question is more 1. if this should be included in core to be "immediately available" and 2. if yes, if the current implementation like naming is ok. And for me both is currently unclear e.g. should we really pick the same naming and accept [potential conflicts in Finland](https://wiki.openstreetmap.org/wiki/Key:hazmat#Finland)?

To merge this PR I would remove it here and if you like, you can submit another PR to clarify the open questions?

Reply:
I think it's a good idea to postpone the addition of the tunnel codes. The Finland issue needs to be sorted out first.